### PR TITLE
feat: update-ember-cli-project

### PR DESCRIPTION
**Pull request by rahulk94**
## Description

This PR bumps this project up to the latest Ember CLI blueprint. This change has primarily been done using [ember-cli-update](https://github.com/ember-cli/ember-cli-update) followed up by commits to update tests and the source to follow newer lint rules.

This should be pretty straightforward / match other Ember CLI updates for addons as I've tried to do keep things as standard as possible here.

Not all lint rules have been addressed at this stage, I think upgrading the CLI version and later addressing lint rules is probably a more tactiful approach.

The tests are passing with Ember@3 LTS versions and failing with Ember@4 versions (which is fine since we're not aiming to add support for that in this PR as it drops lots of deprecated functionality). Embroider safe and optimized builds also fail atm and I'll do a separate PR to address that (as I think there's more issues around Babel with Embroider too).

### New Feature

- [ ] If this is a big feature with breaking changes, consider [opening an issue][issues] to discuss first. This is completely up to you, but please keep in mind that your PR might not be accepted.
- [x] Run unit tests to ensure all existing tests are still passing
- NA - Add new passing unit tests to cover the code introduced by your PR
- NA - Update the readme
- [x] Add some [steps](#steps-to-test) so we can test your cool new feature!
- [x] The PR title is in the [conventional commit](#conventional-commit-spec) format: e.g. `feat(<area>): added new way to do this cool thing #issue-number`
- [x] Add your info to the [contributors](#packagejson-contributors) array in package.json!

## Steps to Test

Running the tests and also the dummy app in this project should work as expected.

I've also integration tested this with our internal Ember app and verified this works as expected. Unfortunately I can't share that as it's closed source.

### CODE REVIEW

File: addon/common/constants.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
-/* global require */
-const APP_VERSION = require('../../package.json').version;
+import { getOwnConfig } from '@embroider/macros';
+const APP_VERSION = getOwnConfig().version;
```
Comment (rahulk94) [inline addon/common/constants.js:2]: Ed F's fix from the Embroider support issue seems to have fixed this when using babel@7 deps
https://github.com/imgix/ember-cli-imgix/issues/210#issuecomment-979689198

File: addon/components/imgix-image-wrapped.js [hunk 2]
```diff
@@ -1,21 +0,0 @@
...
-import layout from '../templates/components/imgix-image-wrapped';
-import ImgixPathBehavior from '../mixins/imgix-path-behavior';
-
-export default Component.extend(ResizeAware, ImgixPathBehavior, {
-  layout: layout,
-  classNames: ['imgix-image-wrap'],
-  init() {
-    this._super(...arguments);
-    deprecate(
-      'imgix-image-wrapped is deprecated and will be removed in ember-cli-imgix@3. Please migrate to using imgix-image instead.',
```
Comment (rahulk94) [inline addon/components/imgix-image-wrapped.js:13]: Please be aware that rather than updating this component I've just chopped it since it is probably best we cut a major now anyway (as the `ember-cli-babel` bump that comes with this PR probably warrants a major)

File: .travis.yml [hunk 3]
```diff
@@ -1,9 +1,11 @@
 language: node_js
 node_js:
-  - lts/*
+  # we recommend testing addons with the same minimum supported node version as Ember CLI
```
Comment (rahulk94) [inline .travis.yml:3]: My understanding is that Travis builds are no longer free. Do we want to migrate this over to GitHub Actions? 

Also if not, I've never touched Travis before so if someone could verify that this YAML file is still valid that would be great.

File: package.json [hunk 4]
```diff
@@ -59,20 +84,22 @@
...
-    "ember-cli-babel": "^6.6.0",
-    "ember-cli-htmlbars": "^2.0.1",
-    "ember-cli-htmlbars-inline-precompile": "^1.0.0",
-    "ember-get-config": "0.2.4",
+    "ember-cli-babel": "^7.26.6",
+    "ember-cli-htmlbars": "^5.7.1",
+    "ember-get-config": "0.2.4 - 0.5.0",
     "ember-resize-aware": "1.2.0",
     "fastboot-transform": "^0.1.2",
     "imgix-core-js": "2.3.2",
```
Comment (rahulk94) [inline package.json:96]: In a separate PR we can switch this to use `imgix/js-core` since thats the new preferred name

**Review by frederickfogerty — COMMENTED** [review]
This looks good to me on the surface. I tried running the tests on CI (excluding lint), and it seems that the ember try cases are failing - even though the basic tests are succeeding. Take a look [here for failing tests](https://app.travis-ci.com/github/imgix/ember-cli-imgix/builds/242719630), and [here for succeeding tests](https://app.travis-ci.com/github/imgix/ember-cli-imgix/builds/242719897)

**Comment by frederickfogerty** [comment]
To be clear here, I'm pretty happy with the state of this PR. If we can get to a resolution about the Travis build and what to do to make sure it passes, then I'm happy to merge this.

**Comment by rahulk94** [comment]
Hey @frederickfogerty thanks the quick first pass and getting the build to run on Travis.

> Take a look here for failing tests
```
=== Scenario: ember-lts-3.20 ===================================================
Result: true

=== Scenario: ember-lts-3.24 ===================================================
Result: true

=== Scenario: ember-lts-3.28 ===================================================
Result: true

=== Scenario: ember-release ====================================================
To use these addons, your app needs ember-auto-import >= 2: ember-source
Result: false

=== Scenario: ember-beta =======================================================
To use these addons, your app needs ember-auto-import >= 2: ember-source
Result: false

=== Scenario: ember-canary =====================================================
To use these addons, your app needs ember-auto-import >= 2: ember-source
=================================================================================
Result: false

=== Scenario: ember-classic ====================================================
Result: true

=== Scenario: embroider-safe ===================================================
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received
The build has been terminated
```

Yeah so I'm not 100% sure what the best approach here is atm the moment. The Ember LTS series against version 3 all pass which is great, but the release, beta and canary are all failing due to us still using [ember-auto-import (EAI)](https://github.com/ef4/ember-auto-import) 1.x. EAI has recently had a major release which is not backwards compatible (I'll let you skim the [upgrade/migration summary section](https://github.com/ef4/ember-auto-import/blob/main/docs/upgrade-guide-2.0.md) rather than try re-explain it) and so that fails those builds. Ember release/beta/canary are all running Ember@4 as well which drops deprecated functionality (I have not checked if we are doing anything thats been dropped just yet). Embroider Ember Try builds are expected to fail at this stage (I haven't tried getting compatibility going and they just timeout weirdly).

Summing up, what are your thoughts on commenting the release, beta, canary, and Embroider options out in the Ember Try config and cutting a 3.0.0-beta.1 style release? I can then look at making Embroider options work since thats what I'm hear to try solve 😂  and do a 3.0.0-beta.2 and then we can see if we want/need to do anything else before cutting a 3.0.0 release?

Ember@4 support can be discussed in a later PR but depending on how easy it is/my capacity I may cheap out and just do a README note and Issue saying Ember we need to add Ember@4 compat soon.

**Comment by frederickfogerty** [comment]
Right, that makes sense as to why those builds are failing then, and I'm happy to comment out those stages and proceed as you suggested. It seems that the ember try environment variable (`EMBER_TRY_SCENARIO`) isn't working properly, I will need to fix this before merging as well. Do you have any ideas how to fix this? I could also just use one `ember try` since it seems that is already doing all of the steps 😅

Totally understand about Ember 4, and thanks for all your help until now!

**Comment by frederickfogerty** [comment]
Right, that makes sense as to why those builds are failing then, and I'm happy to comment out those stages and proceed as you suggested. It seems that the ember try environment variable (`EMBER_TRY_SCENARIO`) isn't working properly, I will need to fix this before merging as well. Do you have any ideas how to fix this? I could also just use one `ember try` since it seems that is already doing all of the steps 😅

Totally understand about Ember 4, and thanks for all your help until now!

**Comment by rahulk94** [comment]
Hey @frederickfogerty I've removed the Ember@4 and Embroider stuff per our earlier discussion.

I've also updated the Travis.yml to call `yarn test:ember-compatibility` in the second stage so it _should_ call Ember Try now, whereas earlier I think it was just doing ember test again 😅 . I reckon start the build and lets see how far we get now.

File: package.json [hunk 5]
```diff
@@ -59,20 +84,21 @@
     "responsive"
   ],
   "dependencies": {
-    "babel-plugin-inline-package-json": "^2.0.0",
-    "babel-plugin-transform-object-rest-spread": "^6.26.0",
```
Comment (rahulk94) [inline package.json:63]: @frederickfogerty I've removed this plugin (and the newer equivalent) in the most recent commit so that consumers of this dependency don't specifically need the version we set. I was seeing some babel weirdness when integrating this with our app and realized its cause we'd declared a version new version of this as the minimum. 

Fwiw its brought in transitively via ember-cli-babel so I reckon we're safe without it?

Comment (frederickfogerty) [reply package.json:63]: Sounds good to me 👍

**Comment by sherwinski** [comment]
> Oh also, commit lint is complaining at me but I can't tell what I've done wrong since it just links to generic docs rather than the broken rule so if you can tell let me know and I can touch it up 😂

@rahulk94 Maybe I can chime in here. Commit lint expects the PR title to conform to a specific format, detailed [here](https://github.com/imgix/ember-cli-imgix/blob/main/.github/pull_request_template.md#conventional-commit-spec). So if you were to edit the PR title, that check should pass.

**Comment by rahulk94** [comment]
@sherwinski 🤦🏾‍♂️ cheers mate... Had a `/` instead of a `:` 🥲

**Comment by frederickfogerty** [comment]
Hey @rahulk94 - just leaving an interim comment here. I worked on Friday to try get the Travis build to pass, but didn't succeed. I will have a bit more time tomorrow, so hopefully I can have an update to you by then 👍  You can track my progress [here](https://app.travis-ci.com/github/imgix/ember-cli-imgix/branches) for the branch "upgrade/update-ember-cli"

**Review by frederickfogerty — APPROVED** [review]
LGTM! Thanks so much for the help here @rahulk94 🙌

**Comment by frederickfogerty** [comment]
I got the tests passing [here](https://app.travis-ci.com/github/imgix/ember-cli-imgix/builds/242977306) so I'll go ahead and merge this 🎉
